### DISCUSSION TITLE
Pi OS Bookworm, Python packages and  PEP 668

### DISCUSSION THREAD

**Discussion by viragored** [GeekFunkLabs/fluidpatcher#86] [General]
I just tried building a "Headless Raspberry Pi Synthesiser" following the instructions on [YouTube](https://www.youtube.com/watch?v=ONuXCzcuj7E). I'm using a new Pi 3A+ and a new SD card.
Pi Imager now offers as default OS the port of Debian Bookworm - the synth script failed.
It seems that Pi Imager now offers only Bullseye as the "Legacy" OS, and the script ran to completion using that. 
Next step for me now is to get something that will send MIDI to the Synth. Fortunately it's not long till my birthday :-)

**Comment by albedozero** [comment]
Thanks for the spot - it looks like this is because Bookworm follows [PEP 668](https://peps.python.org/pep-0668/) and protects the user from installing over system python packages by default. I'll have a fix for this soon.

**Reply by albedozero** [reply]
I made some tweaks to the code so that it relies only on python packages in the Raspberry Pi OS repository (i.e. `sudo apt-get install`-able). I favor this over having the installer create a `venv`, since the fluidpatcher code is installed in its own directory, not site-wide or even in a user site-packages directory. Maybe this should change if this gets put on PyPi, but for now this seems like the right way.
